python/pysrc/kaskada/sources/arrow.py

```diff
@@ -109,15 +105,61 @@ def __init__(
         )
 
         self._convert_options = pyarrow.csv.ConvertOptions(column_types=schema)
-        self.add_rows(rows)
 
-    def add_rows(self, rows: dict | list[dict]) -> None:
+    @staticmethod
+    async def create(
+        *,
+        time_column: str,
+        key_column: str,
+        retained: bool = True,
+        rows: Optional[dict | list[dict]] = None,
+        subsort_column: Optional[str] = None,
+        schema: Optional[pa.Schema] = None,
+        grouping_name: Optional[str] = None,
+        time_unit: Optional[TimeUnit] = None) -> PyDict:
+        """Create a source reading from rows represented as dicts.
+
+        Args:
+            time_column: The name of the column containing the time.
+            key_column: The name of the column containing the key.
+            retained: Whether added rows should be retained for queries.
+              If True, rows (both provided to the constructor and added later) will be retained
+              for interactive queries. If False, rows will be discarded after being sent to any
+              running materializations. Consider setting this to False when the source will only
+              be used for materialization to avoid unnecessary memory usage.
```

Comment:
I think it's good the default is to retain input data, but I wonder if we need to more explicitly say that if you want to run queries over your data, it has to be retained, otherwise we'll discard them. People may associate "retained" as an additional layer of persistence that they shouldn't need for local querying, where they just want to add some rows to in-memory storage then query over them a few times to try things out.